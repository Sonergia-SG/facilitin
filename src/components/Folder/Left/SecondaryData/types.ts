import {
  FolderMOAString,
  FolderMOEString,
  FolderSiteString,
} from '../../../../store/reducer/entities/types';

export interface FormFieldList {
  label: string;
  value: string | null;
  values: Array<{
    key: string;
    label: string;
  }>;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'list';
}

export interface FormFieldDate {
  label: string;
  value: string | null;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'date';
}

export interface FormFieldTextBicRule {
  format: 'bic';
}

export interface FormFieldTextIbanRule {
  format: 'iban';
}

export interface FormFieldTextPhoneRule {
  format: 'phone';
}

export type FormFieldTextRules =
  | FormFieldTextPhoneRule
  | FormFieldTextBicRule
  | FormFieldTextIbanRule;

export interface FormFieldText {
  label: string;
  value: string | null;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'text';
  rules?: FormFieldTextRules;
}

export interface FormFieldNumber {
  label: string;
  value: string | null;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'number';
}

export type FormFieldDef =
  | FormFieldDate
  | FormFieldText
  | FormFieldNumber
  | FormFieldList;

export interface FormSectionDef {
  label: string;
  type: 'section';
  fields: Array<FormFieldDef>;
}

export type FormDef = FormFieldDef | FormSectionDef;

export type GenericForms = Array<FormDef>;

export interface Forms {
  moa: GenericForms;
  moe: GenericForms;
  site: GenericForms;
}
