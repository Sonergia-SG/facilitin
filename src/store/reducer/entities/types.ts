import {
  LOGOUT,
  FOLDER_LOADED,
  LIST_LOADED,
  FOLDER_UPDATE_MOA_LOADED,
  FOLDER_UPDATE_CHECK_POINT_LOADING,
  FOLDER_UPDATE_CHECK_POINT_LOADED,
  FOLDER_UPDATE_CHECK_POINT_ERROR,
  COMMENTS_LIST_LOADED,
  FOLDER_FILE_LITIGE_LOADED,
} from '../../types';

export interface User {
  id_user: number;
  nom: string;
  prenom: string;
}

export interface Users {
  [index: number]: User;
}

export interface CommentSimple {
  id_log: number;
  id_user: number;
  date_log: string;
  message: string;
}

export interface Comment extends CommentSimple {
  user: number;
}

export interface CommentFull extends CommentSimple {
  user: User;
}

export interface Comments {
  [index: number]: Comment;
}

export type BooleanNumber = 0 | 1;

export interface SimpleFile {
  id_dp_file: number;
  id_file: number;
  id_dp_operation: number;
  id_dossierprime: number;
  is_devis: BooleanNumber;
  is_aat: BooleanNumber;
  is_facture: BooleanNumber;
  is_lettre_engagement: BooleanNumber;
  is_justif_preca: BooleanNumber;
  is_autres: BooleanNumber;
  is_aft: BooleanNumber;
  is_contrat_distrib: BooleanNumber;
  is_conv_tepcv: BooleanNumber;
  is_conv_regroupement: BooleanNumber;
  is_liste_depenses: BooleanNumber;
  is_attestation_tepcv: BooleanNumber;
  is_attestation_collectivite: BooleanNumber;
  is_etat_recapitulatif: BooleanNumber;
  is_ah: BooleanNumber;
  is_horodatage: BooleanNumber;
  is_subrogation: BooleanNumber;
  statut: number;
  litige?: BooleanNumber;
}

export interface File extends SimpleFile {}

export interface FileFull extends SimpleFile {}

export interface Files {
  [index: number]: File;
}

export interface MOA {
  moa_prenom: string;
  moa_nom: string;
  moa_fonction: string;
  moa_rue: string;
  moa_rue2: string;
  moa_cp: string;
  moa_ville: string;
  moa_denomination: string;
}

export interface FolderMOAString {
  moa_civilite: string;
  moa_nom: string;
  moa_prenom: string;
  moa_rue: string;
  moa_rue2: string;
  moa_cp: string;
  moa_ville: string;
  moa_pays: string;
  moa_tel: string;
  moa_tel_2: string;
  moa_mobile: string;
  moa_email: string;
  moa_bic: string;
  moa_iban: string;
  moa_siret: string;
  moa_denomination: string;
  moa_fax: string;
  moa_commentaire: string;
  moa_individu_email: string;
  moa_contact: string;
  moa_contact_mobile: string;
  moa_raison_sociale_siege: string;
  moa_siren_siege: string;
  moa_adresse_siege: string;
  moa_cp_siege: string;
  moa_ville_siege: string;
  moa_fonction: string | null;
  moa_individu_fonction: string | null;
  moa_beneficiaire_role: string | null;
}

export interface SimpleFolder extends FolderMOAString {
  id_dp_operation: number;
  id_dossierprime: number;
  code_operation: string;
  moa?: MOA[];
  moe?: {}[];
  travaux?: {}[];
  moa_est_societe: 0;
  moa_is_syndic: BooleanNumber;
  moa_no_siret: BooleanNumber;
  moa_categorie_menage: number;
  moa_nombre_menage: number;
  moa_nombre_personne: number;
  moa_remp_preca: number;
  moe_denomination: string;
  moe_siret: string;
  moe_tel: string;
  moe_fax: string;
  moe_rue: string;
  moe_cp: string;
  moe_ville: string;
  moe_role: number;
  moe_commentaire: string;
  moe_individu_email: string;
  moe_contact: string;
  moe_contact_prenom: string;
  moe_contact_mobile: string;
  moe_contact_fonction: string;
  moe_bic: string;
  moe_iban: string;
  adresse_travaux_rue: string;
  adresse_travaux_nomsite: string;
  adresse_travaux_rue2: string;
  adresse_travaux_cp: string;
  adresse_travaux_ville: string;
  date_debut_travaux: string | null;
  date_fin_travaux: string | null;
  /* code_operation: "BAR-TH-106"
  date_reception: "2018-06-11"
  delai_instruction?: "331 jours de retard"
  id_dossierprime: 19925
  id_dp_operation: 60839
  is_avant_projet: 0
  label_public: "Dossier Incomplet"
  moa_denomination: ""
  moa_nom: "JAUD"
  moa_prenom: "Jean-Marie"
  statut_operation: 13 */
}

export interface Folder extends SimpleFolder {
  documents?: number[];
}

export interface FolderFull extends SimpleFolder {
  documents?: FileFull[];
}

export interface Folders {
  [index: number]: File;
}

export interface CheckPointCategory {
  id_point_controle_categorie: number;
  code_categorie: string;
  nom_categorie: string | null;
}

export interface CheckPointCategories {
  [index: number]: CheckPointCategory;
}

export interface CheckPoint {
  id_point_controle: number;
  id_categorie: number;
  nom: string;
  id_penalite: BooleanNumber;
  automatique: BooleanNumber;
  is_controle_file: BooleanNumber;
  pivot: {
    id_dp_operation: number;
    id_point_controle: number;
    valide: BooleanNumber;
    id_dp_file?: number;
  };
  pointcontrolcategories: CheckPointCategory;
}

export interface CheckPoints {
  [index: number]: CheckPoint;
}

export interface OperationSimple {
  id_dp_operation: number;
  id_dossierprime: number;
  id_operation: number;
  code_operation: string;
  statut?: {
    code_statut?: number;
    label_public?: string;
    id_statut?: number;
    label_prive?: string;
  };
  moderemuneration: {
    id_remuneration: number;
    delai_instruction: string;
  };
}

export interface Operation extends OperationSimple {
  dossierprime?: number;
  dossierprimefile?: [number];
  point_controles: [number];
}

export interface OperationFull extends OperationSimple {
  dossierprime?: FolderFull;
  dossierprimefile?: [FileFull];
  point_controles: [CheckPoint];
}

export interface Operations {
  [index: number]: Operation;
}

export interface Entities {
  files: Files;
  folders: Folders;
  checkPoints: CheckPoints;
  checkPointCategories: CheckPointCategories;
  operations: Operations;
  users: Users;
  comments: Comments;
}

export interface Normalized {
  entities: Entities;
}

export interface FilesFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface FilesFolcerCheckPointLoaded {
  type: typeof FOLDER_UPDATE_CHECK_POINT_LOADED;
  idDpFile: number;
  statusCode: number | null;
}

export interface FileLitigeLoaded {
  type: typeof FOLDER_FILE_LITIGE_LOADED;
  idDpFile: number;
  statusCode: number | null;
}

export interface FilesLogoutAction {
  type: typeof LOGOUT;
}

export type FilesActions = | FilesFolderLoadedAction
| FilesFolcerCheckPointLoaded
| FileLitigeLoaded
| FilesLogoutAction;

export interface FoldersFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface FoldersListLoadedAction {
  type: typeof LIST_LOADED;
  normalized: Normalized;
}

export interface FoldersLogoutAction {
  type: typeof LOGOUT;
}

export interface FoldersUpdateMoaLoaded {
  type: typeof FOLDER_UPDATE_MOA_LOADED;
  id_dossierprime: number;
  values: { [index: string]: string };
}

export type FoldersActions = | FoldersFolderLoadedAction
| FoldersLogoutAction
| FoldersUpdateMoaLoaded
| FoldersListLoadedAction;

export interface CheckPointsFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface CheckPointsListLoadedAction {
  type: typeof LIST_LOADED;
  normalized: Normalized;
}

export interface CheckPointsLogoutAction {
  type: typeof LOGOUT;
}

export interface CheckPointsFolderUpdateCheckpointLoadingAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_LOADING;
  checkPointId: number;
}

export interface CheckPointsFolderUpdateChekpointLoadedAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_LOADED;
  folderId: number;
  checkPointId: number;
}

export interface CheckPointsFolderUpdateCheckpointErrorAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_ERROR;
  checkPointId: number;
  preValue: BooleanNumber;
}

export type CheckPointsActions = | CheckPointsFolderUpdateCheckpointLoadingAction
| CheckPointsFolderUpdateChekpointLoadedAction
| CheckPointsFolderUpdateCheckpointErrorAction
| CheckPointsFolderLoadedAction
| CheckPointsListLoadedAction
| CheckPointsLogoutAction;

export interface CheckPointCategoriesFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface CheckPointCategoriesLogoutAction {
  type: typeof LOGOUT;
}

export type CheckPointCategoriesActions = | CheckPointCategoriesFolderLoadedAction
| CheckPointCategoriesLogoutAction;

export interface OperationsFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}
export interface OperationsListLoadedAction {
  type: typeof LIST_LOADED;
  normalized: Normalized;
}

export interface OperationsLogoutAction {
  type: typeof LOGOUT;
}

export type OperationsActions = | OperationsFolderLoadedAction
| OperationsListLoadedAction
| OperationsLogoutAction;

export interface UsersCommentsListLoadedAction {
  type: typeof COMMENTS_LIST_LOADED;
  normalized: Normalized;
}

export type UsersActions = UsersCommentsListLoadedAction;

export interface CommentsCommentsListLoadedAction {
  type: typeof COMMENTS_LIST_LOADED;
  normalized: Normalized;
}

export type CommentsActions = CommentsCommentsListLoadedAction;
