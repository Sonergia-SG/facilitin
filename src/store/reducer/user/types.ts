import {
  USER_ADD_TOKEN,
  LOGOUT,
  USER_INFOS_LOADING,
  USER_INFOS_LOADED,
  USER_INFOS_ERROR,
  INIT,
} from '../../types';

export type UserFonction = 'instructeur_initial' | 'instructeur_suivi' | '';

export interface UserInfos {
  nom: string;
  prenom: string;
  identifiant: string;
  id_groupe: number;
  fonction: UserFonction;
}

export interface UserState {
  apiKey: string | null;
  infosLoading: boolean;
  user: UserInfos | null;
}

export interface UserInit {
  type: typeof INIT;
}

export interface UserAddTokenAction {
  type: typeof USER_ADD_TOKEN;
  apiKey: string | null;
}

export interface UserInfosLoading {
  type: typeof USER_INFOS_LOADING;
}

export interface UserInfosLoaded {
  type: typeof USER_INFOS_LOADED;
  user: UserInfos;
}

export interface UserInfosError {
  type: typeof USER_INFOS_ERROR;
}

export interface UserLogoutAction {
  type: typeof LOGOUT;
}

export type UserActionTypes = | UserInit
  | UserAddTokenAction
  | UserLogoutAction
  | UserInfosLoading
  | UserInfosLoaded
  | UserInfosError;
