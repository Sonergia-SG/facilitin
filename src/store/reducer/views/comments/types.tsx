import {
  INIT,
  COMMENTS_LIST_LOADING,
  COMMENTS_LIST_LOADED,
  COMMENTS_LIST_ERROR,
  UPDATE_NEW_COMMENT_MESSAGE,
  COMMENTS_POST_COMMENT_LOADING,
  COMMENTS_POST_COMMENT_LOADED,
  COMMENTS_POST_COMMENT_ERROR,
} from '../../../types';
import { CommentsLoadedNormalized } from '../../../actions/views/comments';

export interface CommentsByFoldersPending {
  newMessage: string;
  loading: boolean;
  postLoading: boolean;
}

export interface CommentsByFolders {
  pending: CommentsByFoldersPending;
  comments: Array<number>;
}

export interface CommentsState {
  byFolders: {
    [index: number]: CommentsByFolders;
  };
}

interface CommentsIniAction {
  type: typeof INIT;
}

export interface CommentsListLoadingAction {
  type: typeof COMMENTS_LIST_LOADING;
  idDpFolder: number;
}

export interface CommentsListLoadedAction {
  type: typeof COMMENTS_LIST_LOADED;
  idDpFolder: number;
  normalized: CommentsLoadedNormalized;
}

export interface CommentsListErrorAction {
  type: typeof COMMENTS_LIST_ERROR;
  idDpFolder: number;
}

export interface CommentsUpdateNewMessage {
  type: typeof UPDATE_NEW_COMMENT_MESSAGE;
  message: string;
  idDpFolder: number;
}

export interface CommentsPostCommentLoading {
  type: typeof COMMENTS_POST_COMMENT_LOADING;
  idDpFolder: number;
}

export interface CommentsPostCommentLoaded {
  type: typeof COMMENTS_POST_COMMENT_LOADED;
  idDpFolder: number;
}

export interface CommentsPostCommentError {
  type: typeof COMMENTS_POST_COMMENT_ERROR;
  idDpFolder: number;
}

export type CommentsAction = | CommentsIniAction
| CommentsListLoadingAction
| CommentsListLoadedAction
| CommentsListErrorAction
| CommentsUpdateNewMessage
| CommentsPostCommentLoading
| CommentsPostCommentLoaded
| CommentsPostCommentError;
