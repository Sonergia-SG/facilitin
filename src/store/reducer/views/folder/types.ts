import {
  FOLDER_UPDATE_CHECK_POINT_LOADING,
  FOLDER_UPDATE_CHECK_POINT_LOADED,
  FOLDER_UPDATE_CHECK_POINT_ERROR,
  FOLDER_UPDATE_MOA_VALUE,
  FOLDER_UPDATE_MOE_VALUE,
  FOLDER_LOADING,
  FOLDER_LOADED,
  FOLDER_ERROR,
  LOGOUT,
  FOLDER_CLEAN_MOA_VALUE,
  FOLDER_UPDATE_MOA_ERROR,
  FOLDER_UPDATE_MOA_LOADED,
  FOLDER_UPDATE_MOA_LOADING,
  FOLDER_FILE_LITIGE_LOADING,
  FOLDER_FILE_LITIGE_ERROR,
  FOLDER_FILE_LITIGE_LOADED,
  FOLDER_ENDING_LOADING,
  FOLDER_ENDING_LOADED,
  FOLDER_ENDING_ERROR,
  FOLDER_CLEAN_MOE_VALUE,
  FOLDER_UPDATE_MOE_LOADING,
  FOLDER_UPDATE_MOE_ERROR,
  FOLDER_UPDATE_MOE_LOADED,
} from '../../../types';
import { BooleanNumber } from '../../entities/types';

export type FolderCheckPointStatus = 'SENDING' | 'ERROR';

export interface FolderPendingItem {
  loading?: boolean;
  moaLoading?: boolean;
  moeLoading?: boolean;
  endingLoading?: boolean;
  litigeLoading?: boolean;
  litige?: {
    [index: number]: {
      loading: boolean;
    };
  };
  checkPoint?: {
    [index: number]:
    | {
      status?: FolderCheckPointStatus;
      prevValue?: BooleanNumber;
    }
    | undefined;
  };
  moa?: {
    [index: string]: string;
  };
  moe?: {
    [index: string]: string;
  };
}

export interface FolderState {
  pending: {
    [index: number]: FolderPendingItem | undefined;
  };
}

export interface FolderFolderLoadingAction {
  type: typeof FOLDER_LOADING;
  idDpOperation: number;
}

export interface FolderFolderErrorAction {
  type: typeof FOLDER_ERROR;
  idDpOperation: number;
}

export interface FolderFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  idDpOperation: number;
  normalized: {};
}

export interface FolderFolderUpdateCheckpointLoadingAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_LOADING;
  folderId: number;
  checkPointId: number;
  prevValue: BooleanNumber;
}

export interface FolderFolderUpdateChekpointLoadedAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_LOADED;
  folderId: number;
  checkPointId: number;
}

export interface FolderFolderUpdateCheckpointErrorAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_ERROR;
  folderId: number;
  checkPointId: number;
  prevValue: BooleanNumber;
}

export interface FolderFolderUpdateMoaValue {
  type: typeof FOLDER_UPDATE_MOA_VALUE;
  key: string;
  value: string;
  idDpOperation: number;
}

export interface FolderFolderUpdateMoeValue {
  type: typeof FOLDER_UPDATE_MOE_VALUE;
  key: string;
  value: string;
  idDpOperation: number;
}

export interface FolderFoldercleanMoaValue {
  type: typeof FOLDER_CLEAN_MOA_VALUE;
  idDpOperation: number;
}

export interface FolderFoldercleanMoeValue {
  type: typeof FOLDER_CLEAN_MOE_VALUE;
  idDpOperation: number;
}

export interface FolderLogoutAction {
  type: typeof LOGOUT;
}

export interface FolderFolderUpdateMoaLoading {
  type: typeof FOLDER_UPDATE_MOA_LOADING;
  idDpOperation: number;
}

export interface FolderFolderUpdateMoaError {
  type: typeof FOLDER_UPDATE_MOA_ERROR;
  idDpOperation: number;
}

export interface FolderFolderUpdateMoaLoaded {
  type: typeof FOLDER_UPDATE_MOA_LOADED;
  idDpOperation: number;
}

export interface FolderFolderUpdateMoeLoading {
  type: typeof FOLDER_UPDATE_MOE_LOADING;
  idDpOperation: number;
}

export interface FolderFolderUpdateMoeError {
  type: typeof FOLDER_UPDATE_MOE_ERROR;
  idDpOperation: number;
}

export interface FolderFolderUpdateMoeLoaded {
  type: typeof FOLDER_UPDATE_MOE_LOADED;
  idDpOperation: number;
}

export interface FolderFolderLitigeLoading {
  type: typeof FOLDER_FILE_LITIGE_LOADING;
  idDpOperation: number;
  idDpFile: number;
}

export interface FolderFolderLitigeError {
  type: typeof FOLDER_FILE_LITIGE_ERROR;
  idDpOperation: number;
  idDpFile: number;
}

export interface FolderFolderLitigeLoaded {
  type: typeof FOLDER_FILE_LITIGE_LOADED;
  idDpOperation: number;
  idDpFile: number;
}

export interface FolderFolderEndingLoading {
  type: typeof FOLDER_ENDING_LOADING;
  idDpOperation: number;
}

export interface FolderFolderEndingLoaded {
  type: typeof FOLDER_ENDING_LOADED;
  idDpOperation: number;
}

export interface FolderFolderEndingError {
  type: typeof FOLDER_ENDING_ERROR;
  idDpOperation: number;
}

export type FolderAction = | FolderFolderLoadingAction
| FolderFolderErrorAction
| FolderFolderLoadedAction
| FolderFolderUpdateCheckpointLoadingAction
| FolderFolderUpdateChekpointLoadedAction
| FolderFolderUpdateCheckpointErrorAction
| FolderFolderUpdateMoaValue
| FolderFolderUpdateMoeValue
| FolderFoldercleanMoaValue
| FolderFoldercleanMoeValue
| FolderFolderUpdateMoaLoading
| FolderFolderUpdateMoaLoaded
| FolderFolderUpdateMoaError
| FolderFolderUpdateMoeLoading
| FolderFolderUpdateMoeLoaded
| FolderFolderUpdateMoeError
| FolderFolderLitigeLoading
| FolderFolderLitigeLoaded
| FolderFolderLitigeError
| FolderFolderEndingLoading
| FolderFolderEndingLoaded
| FolderFolderEndingError
| FolderLogoutAction;
