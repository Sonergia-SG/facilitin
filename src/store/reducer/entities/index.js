import { LIST_LOADED } from '../../types';

const initialState = {
  folders: {},
  documents: {},
  moa: {},
  moe: {},
  travaux: {},
};

const entities = (state = initialState, action) => {
  switch (action.type) {
    case LIST_LOADED:
      return {
        ...state,
        folders: {
          ...state.folders,
          ...action.normalized.entities.folders,
        },
      };
    default:
      return state;
  }
};

export default entities;
