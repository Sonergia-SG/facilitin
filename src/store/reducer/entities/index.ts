import { combineReducers } from 'redux';

import folders from './folders';
import checkPoints from './checkPoints';
import files from './files';
import moa from './moa';
import moe from './moe';
import travaux from './travaux';

export default combineReducers({
  folders,
  moa,
  moe,
  travaux,
  checkPoints,
  files,
});
