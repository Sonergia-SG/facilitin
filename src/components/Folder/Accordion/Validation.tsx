import React from 'react';

import { CheckPoint } from '../../../store/reducer/entities/types';
import { addMessageToQueue } from '../../Alert';

interface Props {
  file: { statut: number; id_dp_file: number };
  folderId: number;
  inLitige: any;
  goNext: () => void;
  loading: boolean;
  checkPoints: Array<CheckPoint> | undefined;
}

const getConfig = (checkPoints: Array<CheckPoint>) => {
  if (checkPoints.every(f => f.pivot.valide === 1)) {
    return {
      title: 'Valider',
      type: 'is-success',
      disabled: false,
    };
  }


  if (checkPoints.some(f => f.pivot.valide === 0 && f.id_penalite === 2)) {
    return {
      title: 'Rejeter',
      type: 'is-danger',
      disabled: false,
    };
  }

  if (checkPoints.some(f => f.pivot.valide === 0 && f.id_penalite === 1)) {
    return {
      title: 'Litige',
      type: 'is-warning',
      disabled: false,
    };
  }

  return {
    title: 'Fichier en cours d\'instruction',
    type: 'is-light',
    disabled: true,
  };
};

const Litige = ({
  file, inLitige, folderId, loading, checkPoints, goNext,
}: Props) => {
  const fileCheckPoints = (checkPoints || []).filter(c => c.pivot.id_dp_file === file.id_dp_file);

  const config = getConfig(fileCheckPoints);

  return (
    <button
      type="button"
      style={{ transition: 'background-color 0.4s ease' }}
      className={`button ${config.type} is-pulled-right${
        loading ? ' is-loading' : ''
      }`}
      disabled={config.disabled}
      id="button-litige"
      onClick={() => {
        addMessageToQueue({
          duration: 3000,
          message: 'fake action',
          type: 'warning',
        });
        goNext();
      }}
    >
      {config.title}
    </button>
  );
};

export default Litige;
