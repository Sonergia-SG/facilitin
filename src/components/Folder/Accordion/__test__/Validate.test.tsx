import React from 'react';
import { shallow } from 'enzyme';

import { ValidationComponent as Validation } from '../Validation';
import { CheckPoint } from '../../../../store/reducer/entities/types';

const defaultCheckPoint: CheckPoint = {
  id_point_controle: 0,
  id_categorie: 0,
  nom: 'Checkpoint test',
  id_penalite: 1,
  automatique: 0,
  is_controle_file: 0,
  pivot: {
    id_dp_operation: 0,
    id_point_controle: 0,
    valide: -1,
    id_dp_file: 0,
  },
  pointcontrolcategories: {
    id_point_controle_categorie: 0,
    code_categorie: 'CAT',
    nom_categorie: 'NAME',
  },
};

describe('Litige button', () => {
  it('render in progress', () => {
    const wrapper = shallow(
      <Validation
        file={{ statut: 0, id_dp_file: 0 }}
        folderId={0}
        ending={() => 'Hello'}
        checkPoints={[defaultCheckPoint]}
        locked={false}
        loading={false}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render validate', () => {
    const wrapper = shallow(
      <Validation
        file={{ statut: 0, id_dp_file: 0 }}
        folderId={0}
        ending={() => 'Hello'}
        checkPoints={[{
          ...defaultCheckPoint,
          pivot: { ...defaultCheckPoint.pivot, valide: 1 },
        }]}
        locked={false}
        loading={false}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render locked', () => {
    const wrapper = shallow(
      <Validation
        file={{ statut: 0, id_dp_file: 0 }}
        folderId={0}
        ending={() => 'Hello'}
        checkPoints={[{
          ...defaultCheckPoint,
          pivot: { ...defaultCheckPoint.pivot, valide: 1 },
        }]}
        locked
        loading={false}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render litige', () => {
    const wrapper = shallow(
      <Validation
        file={{ statut: 0, id_dp_file: 0 }}
        folderId={0}
        ending={() => 'Hello'}
        checkPoints={[{
          ...defaultCheckPoint,
          pivot: { ...defaultCheckPoint.pivot, valide: 0 },
        }]}
        locked={false}
        loading={false}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render rejected', () => {
    const wrapper = shallow(
      <Validation
        file={{ statut: 0, id_dp_file: 0 }}
        folderId={0}
        ending={() => 'Hello'}
        checkPoints={[{
          ...defaultCheckPoint,
          id_penalite: 2,
          pivot: { ...defaultCheckPoint.pivot, valide: 0 },
        }]}
        locked={false}
        loading={false}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
