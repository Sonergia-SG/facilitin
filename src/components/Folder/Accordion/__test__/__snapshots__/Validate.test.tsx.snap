// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Litige button render in progress 1`] = `
<button
  className="button is-light is-pulled-right"
  disabled={true}
  id="button-litige"
  onClick={[Function]}
  style={
    Object {
      "transition": "background-color 0.4s ease",
    }
  }
  type="button"
>
  Fichier en cours d'instruction
</button>
`;

exports[`Litige button render litige 1`] = `
<button
  className="button is-danger is-pulled-right"
  disabled={false}
  id="button-litige"
  onClick={[Function]}
  style={
    Object {
      "transition": "background-color 0.4s ease",
    }
  }
  type="button"
>
  Rejeter
</button>
`;

exports[`Litige button render locked 1`] = `
<button
  className="button is-success is-pulled-right"
  disabled={true}
  id="button-litige"
  onClick={[Function]}
  style={
    Object {
      "transition": "background-color 0.4s ease",
    }
  }
  type="button"
>
  Valider
</button>
`;

exports[`Litige button render rejected 1`] = `
<button
  className="button is-warning is-pulled-right"
  disabled={false}
  id="button-litige"
  onClick={[Function]}
  style={
    Object {
      "transition": "background-color 0.4s ease",
    }
  }
  type="button"
>
  Litige
</button>
`;

exports[`Litige button render validate 1`] = `
<button
  className="button is-success is-pulled-right"
  disabled={false}
  id="button-litige"
  onClick={[Function]}
  style={
    Object {
      "transition": "background-color 0.4s ease",
    }
  }
  type="button"
>
  Valider
</button>
`;
